portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,70 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.*;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    ArrayList<Comment> comments = new ArrayList<>();
+    Query query = new Query("Comment");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);   
+    for (Entity entity : results.asIterable()) {
+      Comment comment = new Comment();
+      long id = entity.getKey().getId();
+      comment.commentor = (String) entity.getProperty("name");
+      comment.comment = (String) entity.getProperty("comment");
+      comments.add(comment);
+    }
+    String json = convertToJsonUsingGson(comments);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Comment Tempcomment = new Comment();
+    Tempcomment.commentor = request.getParameter("commentor");
+    Tempcomment.comment = request.getParameter("comment");
+    String json = convertToJsonUsingGson(Tempcomment);
+   // comments.add(Tempcomment);
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("name", Tempcomment.commentor);
+    commentEntity.setProperty("comment", Tempcomment.comment);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
+    response.setContentType("application/json");
+    response.getWriter().println(json);
+
+    response.sendRedirect("/index.html");
+  }
+
+  private String convertToJsonUsingGson(Object obj) {
+    Gson gson = new Gson();
+    String json = gson.toJson(obj);
+    return json;
   }
+
 }
+
+
+class Comment{
```

Comment:
Please make this a separate POJO. That was, it'll be easier to use it as an entity when you integrate with database.